### PR TITLE
properly return null as documented in phpdoc, when synctoken is expired

### PR DESCRIPTION
**Pull request by georgehrke**
- [ ] caldavtester is not happy 😢

### CODE REVIEW

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -1768,7 +1768,7 @@ function getChangesForCalendar($calendarId, $syncToken, $syncLevel, $limit = nul
 		$stmt->execute([ $calendarId ]);
 		$currentToken = $stmt->fetchColumn(0);
 
-		if (is_null($currentToken)) {
+		if ($currentToken === false) {
```
Comment (georgehrke) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: Not sure why this was `is_null`. fetchColumn returns false when there are no more rows:
https://github.com/nextcloud/3rdparty/blob/master/doctrine/dbal/lib/Doctrine/DBAL/Driver/ResultStatement.php#L95

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 2]
```diff
@@ -1780,6 +1780,17 @@ function getChangesForCalendar($calendarId, $syncToken, $syncLevel, $limit = nul
 		];
 
 		if ($syncToken) {
+			// Did the sync token expire?
+			$expireCheckQuery = $this->db->getQueryBuilder();
+			$syncTokenId = $expireCheckQuery->select(['id'])
+				->from('calendarchanges')
+				->where($expireCheckQuery->expr()->eq('synctoken', $expireCheckQuery->createNamedParameter($syncToken)))
+				->execute()
```
Comment (ChristophWurst) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: I prefer to use the query builder like

```php
$query = ...
$result = $query->execute();
... do something with result
$result->closeCursor();
```

Not sure what happens if you don't close the cursor. The db might keep the result set in memory for no reason. cc @nickvergessen

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 3]
```diff
@@ -1780,6 +1780,17 @@ function getChangesForCalendar($calendarId, $syncToken, $syncLevel, $limit = nul
 		];
 
 		if ($syncToken) {
+			// Did the sync token expire?
+			$expireCheckQuery = $this->db->getQueryBuilder();
+			$syncTokenId = $expireCheckQuery->select(['id'])
```
Comment (ChristophWurst) [inline apps/dav/lib/CalDAV/CalDavBackend.php]:
```suggestion
			$syncTokenId = $expireCheckQuery->select('id')
```

File: apps/dav/lib/CalDAV/CalDavBackend.php (hunk 2 above)
Comment (nickvergessen) [reply apps/dav/lib/CalDAV/CalDavBackend.php]: Yeah, it basically keeps a handle open unnecessarily. Default is `number of file handles` so ~1k on default ubuntu. It's unlikely to cause troubles in this case, but in general it should always be done. Just imaging 1k requests being handled at the same time, all have 1 dangling resource...

**Comment by rullzer** [comment]
I'm going to close this due to lack of activity.
Feel free to reopen if anybody wants to continue.

**Pull request closed without merging**